test/server/lib/Archive.ts

```diff
@@ -0,0 +1,74 @@
+import {Archive, create_zip_archive} from 'app/server/lib/Archive';
+import {MemoryWritableStream} from 'app/server/utils/MemoryWritableStream';
+import decompress from 'decompress';
+import {assert} from 'chai';
+import * as stream from 'node:stream';
+
+const testFiles = [
+  {
+    name: "Test1.txt",
+    contents: "This is the contents of a test file.",
+  },
+  {
+    name: "Test2.txt",
+    contents: "This is some other file.",
+  }
+];
+
+async function* generateTestArchiveContents() {
+  yield {
+    name: testFiles[0].name,
+    data: stream.Readable.from(Buffer.from(testFiles[0].contents))
+  };
+  yield {
+    name: testFiles[1].name,
+    data: Buffer.from(testFiles[1].contents)
+  };
+}
+
+async function assertArchiveCompletedSuccessfully(archive: Archive) {
+  await assert.isFulfilled(archive.completed);
+  assert.isTrue(archive.dataStream.closed, "archive data stream was not closed");
+}
+
+async function assertArchiveFailedWithError(archive: Archive) {
+  await assert.isRejected(archive.completed);
+  assert.isTrue(archive.dataStream.closed, "archive data stream was not closed");
+}
+
+async function assertArchiveContainsTestFiles(archiveData: Buffer) {
+  const files = await decompress(archiveData);
+  assert.equal(files.length, testFiles.length);
+
+  for (const testFile of testFiles) {
+    const zippedFile = files.find((file) => file.path === testFile.name);
+    assert.notEqual(zippedFile, undefined);
+    assert.deepEqual(zippedFile?.data.toString(), testFile.contents);
+  }
+}
+
+describe('create_zip_archive', function () {
```

Comment:
Can you wrap everything in a describe Archive? It is helpful heuristically to have a top level test name that matches the filename. Not 100% done in codebase but done enough that it is annoying when not done.